test/framework/assertions.ts

```diff
@@ -0,0 +1,17 @@
+import { expect } from '../utils/expect';
+
+/**
+ * Asserts that a specified field of the actual result is equal to an expected value.
+ * @param actual The result of the previous action.
+ * @param input Specifies both the field and the expected value of the field that
+ *        should be compared with the actual result.
+ */
+export async function assertFieldEqualsAsync(
+    actual: any,
+    input: {
+        field: string;
+        value: any;
+    },
+): Promise<void> {
+    expect(actual[input.field]).to.be.deep.eq(input.value);
```

Comment:
nice!

Reply:
Will `expect` log a helpful error message if it fails?

Reply:
As of now, no. I'll add that as an optional parameter to the `AssertionInfo` interface. I'm going to improve error handling today, which should mitigate this to an extent.